### ISSUE TITLE
Add `unique_id` property to entities

### ISSUE THREAD

**Issue by m01** [asantaga/wiserHomeAssistantPlatform#81]
First of all, thanks for putting together this great integration!

I was trying to assign the thermostat control in the home-assistant UI to an area (the room), so that it's grouped with any other devices in the same room. However, I get the following message when attempting to do so (by going to `Configuration`->`Entities`, and clicking `Wiser <room name>`, the room stat):

> This entity does not have a unique ID, therefore its settings cannot be managed from the UI.

Feature request: I think if e.g. the `WiserRoom` `ClimateDevice` exposed a `unique_id` property, then this should be possible. It's also a [recommended best practice](https://developers.home-assistant.io/docs/integration_quality_scale_index#silver-%F0%9F%A5%88) where possible. Here are some examples from other integrations:
* Tado: https://github.com/home-assistant/core/blob/c8d4cf08d99477bbb8cfb6634697af346ddba65e/homeassistant/components/tado/climate.py#L199
* Hive: https://github.com/home-assistant/core/blob/51ece97e0d2c60c94ba63b25e9207e3f2c2daca2/homeassistant/components/hive/climate.py#L63, which references https://github.com/home-assistant/core/blob/f6fbecf963d91dab9ce44cf26f219c2f53956c2f/homeassistant/components/hive/__init__.py#L185. 
* link to search for others: https://github.com/home-assistant/core/search?p=1&q=unique_id+climatedevice&unscoped_q=unique_id+climatedevice

I suspect it'd be a good idea to expose the `unique_id` property for all of the entities where it's possible, but I'm not 100% sure.

**Comment by asantaga** [comment]
Hey , 

Not against adding a unique ID, (I can use the devices Serial Number for example) but I cant reproduce this issue.

When I go  and look at my devices in configuration/entities  and then select an entity (e.g. the roomstat) it just works..

![image](https://user-images.githubusercontent.com/2940224/77860090-25a20700-7205-11ea-9fc5-c27509aef15e.png)

...
I dont want to implement uniqueids unless I can test it actually fixes the issue

**Comment by msp1974** [comment]
The dev version implements unique ids as part of config flow change.

**Comment by m01** [comment]
How strange. I get this for all the Wiser entities.

![Home assistant wiser missing entity id](https://user-images.githubusercontent.com/783973/77860238-02c42280-7206-11ea-8e11-6d9fff380550.png)

> I dont want to implement uniqueids unless I can test it actually fixes the issue

Of course!

I used the master branch (1.9) and I didn't use the config flow, I configured it using the `configuration.yaml` file. Most home-assistant components' config flow uses auto-discovery, which doesn't generally work with my network setup, but I see yours appears to take a host/ip and secret, so it's probably OK. In any case, I'll give the `dev` branch a go and will report back.

**Comment by asantaga** [comment]
```
The dev version implements unique ids as part of config flow change.
```

Ahhh! thats why I couldnt duplicate it. 

Ive actually pushed 2.0 out, so test that out and I think its all ok.

Angelo

**Comment by asantaga** [comment]
> The dev version implements unique ids as part of config flow change.

I suppose I should have checked :-)

**Comment by m01** [comment]
Using 36a96b8 of this repo with the fix from asantaga/wiserheatingapi#19 assigning areas to the devices now works, thank you!

**Comment by msp1974** [comment]
@m01 the config flow in v2.0 will do one of 3 things.

1) If you have an entry in config.yaml, it will import this into a config flow config.  This is to support all those using prior to v2.0 (like most of us here!).  Once it has done so, entry in config.yaml can be removed (of left alone if you want but it won't use it anymore)/
2) You can manually add by adding through the integration page.  In which case it will ask for ip and secret key and create a config from it.  It has to be able to connect to the hub when creating this way to get the other devices to be created.
3) When this is integrated into the main HA repo, it will support discovery of the hub and just ask for the key.  Zeroconf discovery is only supported for core components.

Hope this helps.  Mark

**Comment by msp1974** [comment]
> How strange. I get this for all the Wiser entities.
> 
> ![Home assistant wiser missing entity id](https://user-images.githubusercontent.com/783973/77860238-02c42280-7206-11ea-8e11-6d9fff380550.png)
> 
> > I dont want to implement uniqueids unless I can test it actually fixes the issue
> 
> Of course!
> 
> I used the master branch (1.9) and I didn't use the config flow, I configured it using the `configuration.yaml` file. Most home-assistant components' config flow uses auto-discovery, which doesn't generally work with my network setup, but I see yours appears to take a host/ip and secret, so it's probably OK. In any case, I'll give the `dev` branch a go and will report back.

@m01 the config flow in v2.0 will do one of 3 things.

1) If you have an entry in config.yaml, it will import this into a config flow config.  This is to support all those using prior to v2.0 (like most of us here!).  Once it has done so, entry in config.yaml can be removed (of left alone if you want but it won't use it anymore)/
2) You can manually add by adding through the integration page.  In which case it will ask for ip and secret key and create a config from it.  It has to be able to connect to the hub when creating this way to get the other devices to be created.
3) When this is integrated into the main HA repo, it will support discovery of the hub and just ask for the key.  Zeroconf discovery is only supported for core components.

Hope this helps.  Mark

**Comment by asantaga** [comment]
resolved in 2.1
